src/features/smartSearch/components/inputs/StyledAutocomplete.tsx

```diff
@@ -0,0 +1,448 @@
+    details?: AutocompleteChangeDetails<Item>
+  ) => void;
+  value?: string | number;
+};
+
+const StyledGroupedSelect: FC<Props> = (props) => {
+  const options: Item[] = useMemo(
+    () =>
+      props.items
+        .sort((item0, item1) => {
+          const group0 = item0.group ?? '';
+          const group1 = item1.group ?? '';
+
+          if (group0 === 'pinned') {
+            return -1;
+          }
+
+          if (group1 === 'pinned') {
+            return 1;
+          }
+
+          const ret = group0.localeCompare(group1);
+          if (ret === 0) {
+            return item0.label.localeCompare(item1.label);
+          }
+
+          return ret;
+        })
+        .map((item) => {
+          if (item.group === 'pinned') {
+            return {
+              id: item.id,
+              label: item.label,
+            };
+          }
+          return item;
+        })
+        .map((item) => ({
+          ...item,
+          label: item.label.trim(),
+        })),
+    [props.items]
+  );
+
+  const valueItem = useMemo(
+    () =>
+      props.value === undefined
+        ? props.items.length === 0
+          ? undefined
+          : props.items[0]
+        : options.find((item) => item.id === props.value),
+    [props.value, options]
+  );
+
+  // Use react-window v2's useListRef hook for imperative API access
+  const internalListRef = useListRef(null);
+  const optionIndexMapRef = React.useRef<Map<string, number>>(new Map());
+
+  const handleItemsBuilt = React.useCallback(
+    (optionIndexMap: Map<string, number>) => {
+      optionIndexMapRef.current = optionIndexMap;
+    },
+    []
+  );
+
+  // Handle keyboard navigation by scrolling to highlighted option
+  const handleHighlightChange = (
+    _event: React.SyntheticEvent,
+    option: Item | null
+  ) => {
+    if (option && internalListRef.current) {
+      const index = optionIndexMapRef.current.get(option.id.toString());
+      if (index !== undefined) {
+        internalListRef.current.scrollToRow({ align: 'auto', index });
+      }
+    }
+  };
+
+  const [editMode, setEditMode] = useState(false);
+  const [inputValue, setInputValue] = useState('');
+
+  const onChange = useCallback(
+    (
+      event: React.SyntheticEvent,
+      value: AutocompleteValue<Item, false, true, false>,
+      reason: AutocompleteChangeReason,
+      details?: AutocompleteChangeDetails<Item>
+    ) => {
+      props.onChange?.(
+        {
+          ...event,
+          target: {
+            ...event.target,
+            value: value ? value.id.toString() : '',
+          },
+        },
+        value,
+        reason,
+        details
+      );
+      setInputValue('');
+    },
+    [props.onChange, setInputValue]
+  );
+
+  if (!valueItem) {
+    return null;
+  }
+
+  return (
+    <Autocomplete
+      disableClearable={true}
+      disableListWrap
+      groupBy={(option) => option.group ?? ''}
+      inputValue={inputValue}
+      onChange={onChange}
```

Comment:
Maybe checking (here or elswherere) if `reason ` for the `onChange` is `selectOption` and if so, setting `editMode` to false would be a way to address what i experienced in the film?